File: qc-core/src/main/java/com/eager/questioncloud/core/domain/hub/board/repository/PostRepository.java
package com.eager.questioncloud.core.domain.hub.board.repository;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.hub.board.dto.PostDto.PostDetail;
import com.eager.questioncloud.core.domain.hub.board.dto.PostDto.PostListItem;
import com.eager.questioncloud.core.domain.hub.board.model.Post;
import java.util.List;

public interface PostRepository {
    List<PostListItem> getPostList(Long questionId, PagingInformation pagingInformation);

    List<PostListItem> getCreatorPostList(Long creatorId, PagingInformation pagingInformation);

    int countCreatorPost(Long creatorId);

    PostDetail getPostDetail(Long postId);

    Post findByIdAndWriterId(Long postId, Long userId);

    Post findById(Long postId);

    int count(Long questionId);

    Post save(Post post);

    void delete(Post post);
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/post/infrastructure/repository/PostRepository.kt
package com.eager.questioncloud.core.domain.post.infrastructure.repository

import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.post.model.Post

interface PostRepository {
    fun findByQuestionIdWithPagination(questionId: Long, pagingInformation: PagingInformation): List<Post>

    fun findByQuestionIdInWithPagination(questionIds: List<Long>, pagingInformation: PagingInformation): List<Post>

    fun countByQuestionIdIn(questionIds: List<Long>): Int

    fun findByIdAndWriterId(postId: Long, userId: Long): Post

    fun findById(postId: Long): Post

    fun countByQuestionId(questionId: Long): Int

    fun save(post: Post): Post

    fun delete(post: Post)
}
